src/plugins/twitter/client.py

```diff
@@ -0,0 +1,141 @@
+from datetime import datetime, timedelta
+import requests
+from requests_oauthlib import OAuth1
+
+
+class TwitterClient:
+
+    """ Twitter client for searching tweets about DOIs.
+
+    Please note, this uses the Premium Twitter search API. For more details
+    visit https://developer.twitter.com/en/docs/tweets/search/api-reference
+    /premium-search.html
+    """
+
+    verify_url = 'https://api.twitter.com/1.1/account/verify_credentials.json'
+    search_url_base = 'https://api.twitter.com/1.1/tweets/search'
+
+    def __init__(
+            self,
+            app_key,
+            app_key_secret,
+            access_token,
+            access_token_secret,
+            label
+
+    ):
+        """ Instantiate the Twitter client.
+
+        Args:
+            app_key (str): key for your twitter app
+            app_key_secret (str): key secret for your twitter app
+            access_token (str): access token for your twitter app
+            access_token_secret (str): access token secret for your twitter app
+            label (str): label for your full-archive twitter dev environment
+        """
+
+        self.app_key = app_key
+        self.app_key_secret = app_key_secret
+        self.access_token = access_token
+        self.access_token_secret = access_token_secret
+
+        self.search_url = f'{self.search_url_base}/fullarchive/{label}.json'
+        self.auth = self.set_authentication()
+
+    def set_authentication(self):
+        """ Set authentication credentials used for API calls."""
+        return OAuth1(
+            self.app_key,
+            self.app_key_secret,
+            self.access_token,
+            self.access_token_secret
+        )
+
+    def verify_credentials(self):
+        """ Test credentials to see if they are valid.
+
+        Returns:
+            Response: API response with your user account details or an error.
+        """
+        return requests.get(self.verify_url, auth=self.auth)
+
+    @staticmethod
+    def twitter_date_format(datetime_obj):
+        """ Convert datetime object to a string, in twitter API format."""
+        return datetime_obj.strftime('%Y%m%d%H%M')
+
+    @staticmethod
+    def get_end_date(start_datetime_obj, days=30):
+        """ Add a specific number of days to a given a start datetime.
```

Comment:
I think this docstring is a bit inconsistent. A method called `get` doesn't really express the fact that we are trying to `add` days to a given datetime.

Reply:
Hmmm, True. I'll think of something better :+1: